### ISSUE TITLE
error: expected slot materialize_% to be inactive, is active

### ISSUE THREAD

**Issue by def-** [MaterializeInc/materialize#24357]
### What version of Materialize are you using?

5ec8eb4d8d53

### What is the issue?

Seen in Postgres CDC tests in https://buildkite.com/materialize/tests/builds/72831#018cf6cd-5a22-4c36-8192-cf50fa7eccd7
```
^^^ +++
pg-cdc.td:152:1: error: expected slot materialize_% to be inactive, is active
     |
  14 | > CREATE SECRET pgpa ... [rest of line truncated for security]
  20 |     PASSWORD SECRET  ... [rest of line truncated for security]
  29 |     PASSWORD SECRET  ... [rest of line truncated for security]
  34 | $ postgres-execute c ... [rest of line truncated for security]
 151 | #
 152 | $ postgres-verify-sl ... [rest of line truncated for security]
     | ^
```
I haven't seen this failure before.

ci-regexp: to be inactive, is active
ci-apply-to: pg-cdc

**Comment by nrainer-materialize** [comment]
Further occurrence: https://buildkite.com/materialize/tests/builds/72839#018cf7a5-a7f1-4ea7-940e-f1285977da2c

**Comment by def-** [comment]
I think this might come from https://github.com/MaterializeInc/materialize/pull/24161

**Comment by sploiselle** [comment]
The issue here is that sources running on the default cluster don't signal to the upstream PG cluster that the replication slot is inactive as quickly as those whose clusters drop. This means we are more likely to leak replication slots for dropped sources if they run on the default cluster. I need some time to see if there's a better way to solve this (e.g. shortening the timeout from which PG should hear from us) rather than just lengthening the test timeout + the time we spend trying to drop the replication slot.

**Comment by guswynn** [comment]
This happened again: https://buildkite.com/materialize/tests/builds/73204#018d148c-a028-46b6-9d36-896597bebe0b

**Comment by guswynn** [comment]
Is it possible that this is a race like: https://github.com/MaterializeInc/materialize/issues/24396 ?

**Comment by sploiselle** [comment]
@guswynn I don't think so because the prior test awaits the replication slot to be created, so it's very unlikely we're allowing ourselves to recreate it after dropping it.

**Comment by sploiselle** [comment]
The cause of this issue is a well known and understood race condition in source rendering. @petrosagg has designed a solution to fix the issue [here](https://docs.google.com/document/d/19i1lBaTKY4-G9Vi2_2-FpRzA1RecYuOCcOk0KMnDacI/edit#heading=h.a3wek7c0fc48), though I'm not sure if there's anywhere on GitHub to link to.

Here's the history where this occurs:
1. Testdrive executes `test/pg-cdc/alter-source.td`
1. Creates `mz_source_too`, which sends a command to create an ingestion.
1. Perform some other operations that succeed very quckly.
1. Drop `mz_source_too`, which drops its replication slot––only the replication slot does not exist.
1. The command to create the source gets scheduled on the Timely cluster.
1. When rendering the dataflow, it creates the replication slot.
1. Tests continue to execute until we check for active replication slots in `pg-cdc.td` and fail.

@guswynn was right that this is the same issue at #24396; I had originally mistakenly assumed it was a test that had executed more recently.

To resolve the CI flakes, I'm just going to disable this check for now.
